apis/v1alpha2/httproute_types.go

```diff
@@ -641,6 +657,27 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// Protocol is the protocol to be used in the value of the `Location`
+	// header in the response.
+	// When empty, the protocol of the request is used.
+	// +kubebuilder:validation:Enum=HTTP;HTTPS
```

Comment:
Nit: prefer to leave a blank like between the comment and the codegen tags. Just makes is a litle easier to read.